packages/framer-motion/src/animation/use-animated-state.ts

```diff
@@ -59,14 +58,16 @@ export function useAnimatedState(initialState: any) {
 
     useEffect(() => {
         element.mount({})
-        return element.unmount()
-    }, [])
+        return element.unmount
```

Comment:
This isn't React 18 specific, but something that must have been a quiet bug before